src/main/java/grocery/Grocery.java

```diff
@@ -40,18 +40,20 @@ public void setName(String name) {
     }
 
     public void setAmount(String amount) {
+        assert !(amount.isEmpty()) : "Amount entered is invalid!";
         this.amount = amount;
     }
 
     public void setExpiration(String expiration) {
+        assert !(expiration.isEmpty()) : "Expiration date entered is invalid!";
         this.expiration = expiration;
     }
 
     /**
      * Returns a String representation of the Grocery.
      */
     public String printGrocery() {
-        assert !(this.name.isEmpty()) : "Grocery does not exist";
+        assert !(this.name.isEmpty()) : "Grocery does not exist!!";
```

Comment:
Good addition of assertions to check the user input. It enhances the robustness of command parsing.